python/tests/tornado/test_hyperbahn.py

```diff
@@ -60,5 +61,6 @@ def test_request():
 def test_register():
     channel = TChannel(name='test')
 
-    with pytest.raises(ConnectionClosedError):
+    with patch('tchannel.tornado.hyperbahn.log') as mock_logger:
         yield hyperbahn.advertise(channel, 'foo', ['127.0.0.1:23000'])
+    assert mock_logger.error.called
```

Comment:
Add a proper test for the retry logic. Don't test logging. Test that the send() function was called twice the register payload.